Soft-skip tags with no changelog entry in create-github-release

## Description

When `release-kit create-github-release` is invoked with `--tags` and any requested tag has no changelog entry, the command exits 1 — failing the calling CI workflow. This breaks the `create-github-release.reusable.yaml` workflow whenever a pushed release tag has no changelog content (e.g., a tooling-only release whose changelog generator omitted the entry).

Observed in `williamthorsen/devtools.afg` workflow run 25360510681. A push of tag `devtools-panel-v0.12.1` produced:

```
Warning: no changelog entry for version 0.12.1; skipping GitHub Release creation
Error: requested tags have no changelog entry: devtools-panel-v0.12.1. Verify the tag names match a published changelog version.
Error: Process completed with exit code 1.
```

The release had no audience-relevant content, so the changelog generator legitimately omitted an entry for it. The workflow should silently skip in that case, not fail.

## Context

PR #347 (commit `5e674139`, *release-kit|fix: Skip tooling-only releases instead of failing*) introduced this behavior. It split the `CreateReleaseSkipReason` discriminant in `createGithubRelease` into three:

- `no-entry` — changelog file missing, JSON unparseable, or version not in changelog. **Currently hard-fails when `--tags` is specified.**
- `no-audience-content` — entry exists but has no all-audience sections. Soft-skips.
- `empty-body` — rendered body is empty. Soft-skips.

The justification for keeping `no-entry` as a hard-fail (in `createGithubReleaseCommand.ts`, lines 50-59) was typo protection: a `--tags` value with no matching changelog entry might indicate a misspelled tag name worth surfacing.

That justification doesn't hold. `resolveCommandTags` (`packages/release-kit/src/resolveCommandTags.ts`, lines 64-72) already validates each `--tags` value against the tags resolved from HEAD and exits 1 with `Error: Unknown tag "..."`. By the time a tag reaches `createGithubReleases`, its existence in git is guaranteed — typos cannot reach the `no-entry` branch.

The actual sub-cases of `no-entry` that pass that validation are:

1. The workspace's changelog file is missing (rare misconfig).
2. The changelog JSON is unparseable (broken config).
3. The version is not in the changelog (legitimate tooling-only release where the generator dropped an entry). **This is the case affecting the cited workflow.**

PR #347 already established the principle that "no content to release-note" is a normal outcome (via `no-audience-content` and `empty-body` soft-skips). The same logic should apply when the same underlying condition reaches us via the `no-entry` path.

## Approaches considered

- **A. Treat `no-entry` as a soft-skip uniformly (chosen).** Drop the `--tags` + `no-entry` hard-fail. All three skip reasons behave identically when invoked with `--tags`. The existing `console.warn` calls in `createGithubRelease` already name each cause in logs, and the existing summary `console.info` (`Skipped N tag(s) with no releasable content: <tag> (<reason>), ...`) already lists every skipped tag with its reason.
- **B. Sub-discriminate `no-entry`.** Hard-fail on "file missing" or "JSON unparseable"; soft-skip on "version not in changelog". Adds a `CreateReleaseSkipReason` sub-discriminant for marginal benefit; the misconfig cases are already easy to diagnose from the existing warnings.
- **C. Add a `--strict` opt-in flag.** Default to soft-skip; `--strict` retains today's hard-fail. Adds API surface for behavior nobody currently asks for.
- **D. Patch only the workflow.** Pre-check changelog presence in `create-github-release.reusable.yaml`. Distributes logic and leaves the CLI exposed in other call sites.

## Proposed solution (Option A)

In `packages/release-kit/src/createGithubReleaseCommand.ts`:

1. Delete the conditional block that exits 1 for `no-entry` skips when `--tags` was specified (currently lines 50-59).
2. Update the surrounding comment to record the new invariant: typo protection lives in `resolveCommandTags`, so this command treats every per-tag skip reason as informational. Cross-reference `resolveCommandTags.ts` so a future reader can find that protection.
3. Leave the trailing `console.info` summary (currently lines 61-64) unchanged — it already lists every skipped tag with its reason.

In `packages/release-kit/src/__tests__/createGithubReleaseCommand.unit.test.ts`:

1. Replace the test *exits with code 1 when --tags is explicit and any requested tag has no changelog entry* (lines 202-231) with a test asserting:
   - The command does not exit.
   - `console.error` is not called.
   - `console.info` is called with the formatted summary listing each `no-entry` tag.
2. Replace the test *exits 1 when --tags has mixed outcomes including a no-entry skip, naming only the no-entry tags* (lines 263-291) with a test asserting:
   - The command does not exit.
   - The summary lists both created and skipped tags appropriately (mirroring the existing pattern at lines 307+).
3. Verify that the existing tests for `no-audience-content` (lines 233-247) and `empty-body` (lines 249-261) still pass unchanged.

## Acceptance criteria

### Must have

- `release-kit create-github-release --tags <tag>` exits 0 with a `Skipped …` info summary when the requested tag has skip reason `no-entry`.
- The `create-github-release.reusable.yaml` workflow no longer fails when a pushed tag has no changelog entry.
- The existing soft-skip behavior for `no-audience-content` and `empty-body` is preserved.
- The existing typo protection in `resolveCommandTags` (unknown `--tags` value → exit 1 with `Error: Unknown tag …`) is preserved.
- Unit tests cover the soft-skip behavior for all three skip reasons under `--tags`, and for mixed-outcome cases.

### Should have

- The summary log message continues to name each skipped tag with its reason (already true; preserve).
- The comment in `createGithubReleaseCommand.ts` explains why no per-reason discrimination is needed at this layer (typo protection lives upstream in `resolveCommandTags`).

### Nice to have

- A brief mention in the PR body that this completes the work begun in PR #347 by closing the `no-entry` gap.

--- a/packages/release-kit/src/__tests__/createGithubReleaseCommand.unit.test.ts
+++ b/packages/release-kit/src/__tests__/createGithubReleaseCommand.unit.test.ts
@@ -199,7 +199,10 @@ describe(createGithubReleaseCommand, () => {
     expect(console.error).toHaveBeenCalledWith('Error discovering workspaces: discovery failed');
   });
 
-  it('exits with code 1 when --tags is explicit and any requested tag has no changelog entry', async () => {
+  it('does not exit when --tags is explicit and every skip is no-entry', async () => {
+    // Typo protection lives upstream in resolveCommandTags (which exits 1 on unknown --tags
+    // values), so a no-entry skip reaching here is a legitimate "no releasable content"
+    // outcome — same as no-audience-content and empty-body.
     mockDiscoverWorkspaces.mockResolvedValue(['packages/core', 'packages/extra']);
     mockResolveReleaseTags.mockReturnValue([
       { tag: 'core-v1.3.0', dir: 'core', workspacePath: 'packages/core' },
@@ -213,20 +216,11 @@ describe(createGithubReleaseCommand, () => {
       ],
     });
 
-    let thrown: ExitError | undefined;
-    try {
-      await createGithubReleaseCommand(['--tags=core-v1.3.0,extra-v0.1.0']);
-    } catch (error: unknown) {
-      if (error instanceof ExitError) {
-        thrown = error;
-      }
-    }
+    await createGithubReleaseCommand(['--tags=core-v1.3.0,extra-v0.1.0']);
 
-    expect(thrown).toBeInstanceOf(ExitError);
-    expect(thrown?.code).toBe(1);
-    expect(console.error).toHaveBeenCalledWith(
-      'Error: requested tags have no changelog entry: core-v1.3.0, extra-v0.1.0. ' +
-        'Verify the tag names match a published changelog version.',
+    expect(console.error).not.toHaveBeenCalled();
+    expect(console.info).toHaveBeenCalledWith(
+      'Skipped 2 tag(s) with no releasable content: core-v1.3.0 (no-entry), extra-v0.1.0 (no-entry).',
     );
   });
 
@@ -260,9 +254,9 @@ describe(createGithubReleaseCommand, () => {
     expect(console.info).toHaveBeenCalledWith('Skipped 1 tag(s) with no releasable content: core-v1.3.0 (empty-body).');
   });
 
-  it('exits 1 when --tags has mixed outcomes including a no-entry skip, naming only the no-entry tags', async () => {
-    // A typoed tag is still a typo even when other requested tags succeed. The error message
-    // names only the no-entry tags so the user knows which to investigate.
+  it('logs an info summary when --tags has mixed outcomes including a no-entry skip', async () => {
+    // Mirrors the no-audience-content mixed-outcome test below: no-entry is no longer
+    // discriminated at this layer because resolveCommandTags already rejects unknown tags.
     mockDiscoverWorkspaces.mockResolvedValue(['packages/core', 'packages/extra']);
     mockResolveReleaseTags.mockReturnValue([
       { tag: 'core-v1.3.0', dir: 'core', workspacePath: 'packages/core' },
@@ -273,21 +267,10 @@ describe(createGithubReleaseCommand, () => {
       skipped: [{ tag: 'extra-v0.1.0', reason: 'no-entry' }],
     });
 
-    let thrown: ExitError | undefined;
-    try {
-      await createGithubReleaseCommand(['--tags=core-v1.3.0,extra-v0.1.0']);
-    } catch (error: unknown) {
-      if (error instanceof ExitError) {
-        thrown = error;
-      }
-    }
+    await createGithubReleaseCommand(['--tags=core-v1.3.0,extra-v0.1.0']);
 
-    expect(thrown).toBeInstanceOf(ExitError);
-    expect(thrown?.code).toBe(1);
-    expect(console.error).toHaveBeenCalledWith(
-      'Error: requested tags have no changelog entry: extra-v0.1.0. ' +
-        'Verify the tag names match a published changelog version.',
-    );
+    expect(console.error).not.toHaveBeenCalled();
+    expect(console.info).toHaveBeenCalledWith('Skipped 1 tag(s) with no releasable content: extra-v0.1.0 (no-entry).');
   });
 
   it('does not exit when --tags is omitted and every tag is skipped', async () => {

--- a/packages/release-kit/src/createGithubReleaseCommand.ts
+++ b/packages/release-kit/src/createGithubReleaseCommand.ts
@@ -41,23 +41,11 @@ export async function createGithubReleaseCommand(argv: string[]): Promise<void> 
     process.exit(1);
   }
 
-  // Fail visibly when the user explicitly requested tags via --tags and any requested tag has no
-  // changelog entry: that's a typo or misconfiguration worth surfacing, even if other requested
-  // tags succeeded. Other skip reasons (`no-audience-content`, `empty-body`) are intentional
-  // outcomes of release-kit's rendering rules — a tooling-only release has nothing user-facing
-  // to announce, and that's a normal state, not a failure. Omitted --tags (operate on all HEAD
-  // tags) keeps the legacy soft-skip behavior since the user did not single out specific tags.
-  if (requestedTags !== undefined) {
-    const noEntryTags = outcome.skipped.filter((s) => s.reason === 'no-entry').map((s) => s.tag);
-    if (noEntryTags.length > 0) {
-      console.error(
-        `Error: requested tags have no changelog entry: ${noEntryTags.join(', ')}. ` +
-          `Verify the tag names match a published changelog version.`,
-      );
-      process.exit(1);
-    }
-  }
-
+  // Treat every per-tag skip reason (`no-entry`, `no-audience-content`, `empty-body`) as
+  // informational. Typo protection for `--tags` lives upstream in `resolveCommandTags`, which
+  // exits 1 with `Error: Unknown tag "..."` before any tag reaches `createGithubReleases`.
+  // By the time a tag's outcome is `no-entry` here, its existence in git is guaranteed and the
+  // missing entry is a legitimate "no releasable content" outcome — same as the other reasons.
   if (outcome.skipped.length > 0) {
     const formatted = outcome.skipped.map((s) => `${s.tag} (${s.reason})`).join(', ');
     console.info(`Skipped ${outcome.skipped.length} tag(s) with no releasable content: ${formatted}.`);

